### PR TITLE
Support creating templates only using tags

### LINKED ISSUE #10: New templates notebook gets created everytime
New templates notebook gets created every time when a user installs a plugin in a new desktop app. An dialog could be added to ask user for an already existing templates notebook

### LINKED ISSUE #12: Using a notebook instead of a folder to store the templates can cause various issues
I've been thinking about how we'd now be using a notebook to store the templates, instead of a folder on the hard drive. I can think of many possible issues with this:

- People will delete the template folder. How does your code handle this?

- When setting up a new client, your plugin asks to select a template folder but it hasn't been synced yet.

- Since it's not present, they might ask your plugin to create it and end up with multiple template notebooks each used by a different device.

- People will export the template folder as jex, then import it in a new client, and wonder why your plugin is not using that folder

- They will manually create a folder named "Templates" and wonder why your plugin doesn't pick it up.

- Users will share the Templates notebook using Joplin Server, but the recipient won't be able to use the templates.

I think there are ways to improve this but overall I'm not sure using a template notebook is a good solution. The main problem is that we have no concept of special setting notebooks (there's a thread about this on the forum), which means users can do anything with the notebook and break your plugin behaviour very easily.

So I'm not sure what's your take on this but I get the feeling we've changed a simple solution with no ambiguity (a simple "templates" folder) to one that's quite fragile.

### PR DESCRIPTION
**Pull request by nishantwrp**
This PR does two things.

- Allow creating templates using a tag.
- Remove templates notebook functionality. 

Fixes #10, Fixes #12.

### CODE REVIEW

**Comment by CalebJohn** [comment]
Is this PR still a draft? My templates aren't migrated in to the new system. Also, the popup dialogue says that there is a help option under Templates, but it doesn't seem to exist yet.

**Comment by nishantwrp** [comment]
> My templates aren't migrated in to the new system.
I'll create another PR for this.

> Also, the popup dialogue says that there is a help option under Templates, but it doesn't seem to exist yet.
That will also be done in a separate PR. But wrote this so, I don't have to change it in future.

**Review by CalebJohn — APPROVED** [review]

**Comment by CalebJohn** [comment]
I haven't merged this yet so that we can wait for the conclusion of #12.

**Comment by nishantwrp** [comment]
@CalebJohn please take a look.

File: src/utils/folders.ts [hunk 1]
```diff
@@ -21,29 +12,5 @@ export const doesFolderExist = async (folderId: string): Promise<boolean> => {
 };
 
 export const getAllNotesInFolder = async (folderId: string): Promise<Note[]> => {
-    let pageNum = 1;
```
Comment (CalebJohn) [inline src/utils/folders.ts:19]: Is this function (and module) needed anymore?

**Review by CalebJohn — APPROVED** [review]

**Comment by CalebJohn** [comment]
I just have the one comment, I don't think that function is needed anymore, but it's something you can address later, I'll merge this now. It looks good!

File: src/utils/folders.ts (hunk 1 above)
Comment (nishantwrp) [reply src/utils/folders.ts:19]: No, they're not thanks for pointing that out. If this is the only review comment, you can merge this pr, I'll remove these functions in the next pr.

Comment (CalebJohn) [reply src/utils/folders.ts:19]: Good to see that we're on the same page. I'm looking forward to the next PR.

**Pull request merged**